internal/zinc-classpath/src/main/scala/sbt/internal/inc/ScalaInstance.scala

```diff
@@ -167,7 +167,7 @@ object ScalaInstance {
           case p: ScalaProvider2 @unchecked => Option((provider.loader, p.loaderLibraryOnly))
         }
       } catch {
-        case _: NoSuchMethodError => None
+        case _: ReflectiveOperationException => None
```

Comment:
This includes `ClassNotFoundException, IllegalAccessException, InstantiationException, InvocationTargetException, NoSuchFieldException, NoSuchMethodException`. I think it would be better that we don't have a catch-all handler, and we rather focus on `NoSuchMethodException` only. If there's a `ClassNotFoundException` or `IllegalAccessException` going on, then isn't that a legit issue we want to surface?

Reply:
I guess so. I figured the clarity of `ReflectiveOperationException` might be a tradeoff worth taking, but I can switch to `NoSuchMethodException `.